dev.hdos.HDOS.json

```diff
@@ -0,0 +1,90 @@
+{
+    "id": "dev.hdos.HDOS",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "hdos",
+    "separate-locales": false,
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.openjdk11"
+    ],
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--env=JAVA_HOME=/app/jre",
+        "--talk-name=org.freedesktop.Notifications",
+        "--filesystem=xdg-run/app/com.discordapp.Discord:create",
+        "--filesystem=~/hdos:create",
+        "--filesystem=~/jagex_runescape_preferences.dat:create",
+        "--filesystem=~/jagexcache:create"
```

Comment:
It's an antipattern to store any files directly in user's home directory. Ideally, you should use `XDG_CONFIG_HOME` for configuration, `XDG_DATA_HOME` for "assets", and 'XDG_CACHE_HOME` for cache. Are these three paths configurable somehow?

Reply:
Ideally, we would want to keep the Config in the home/HDOS directory to maintain setting parity between the flatpak version and the jar version and also to make it easier to modify by the end user if desired.

Reply:
> It's an antipattern to store any files directly in user's home directory. Ideally, you should use `XDG_CONFIG_HOME` for configuration, `XDG_DATA_HOME` for "assets", and `XDG_CACHE_HOME` for cache. Are these three `--filesystem` paths configurable somehow?

It's a legacy thing, if the user has the old settings it can read and migrate them. It's been moved. Maybe a middle ground is to make it read only?

Reply:
I'd just document somewhere how to migrate data instead. `XDG_CONFIG` variables map to various subdirectories of `~/.var/app/ID/` so it's not exactly difficult to modify either.

Reply:
@barthalion Ok, I cleaned up the home directory. Strangler binaries should be persisted, but I am a bit suspicious about if the "--persist=" flag also retains temporary files? If it's not advised, is it possible to persist a specific file vs a directory? I'm suspecting it'll be best to just migrate all the legacy binaries into the "hdos" home directory.